driver-core/src/main/java/com/datastax/driver/core/CBUtil.java

```diff
@@ -117,10 +118,47 @@ public static void writeBytes(byte[] bytes, ByteBuf cb) {
         cb.writeBytes(bytes);
     }
 
+    public static void writeBytes(ByteBuffer bytes, ByteBuf cb) {
+        cb.writeShort(bytes.remaining());
+        cb.writeBytes(bytes.duplicate());
+    }
+
     public static int sizeOfBytes(byte[] bytes) {
         return 2 + bytes.length;
     }
 
+    public static int sizeOfBytes(ByteBuffer bytes) {
+        return 2 + bytes.remaining();
+    }
+
+    public static Map<String,ByteBuffer> readBytesMap(ByteBuf cb) {
+        int length = cb.readUnsignedShort();
+        ImmutableMap.Builder<String,ByteBuffer> builder = ImmutableMap.builder();
+        for (int i = 0; i < length; i++) {
+            String k = readString(cb);
+            ByteBuffer v = ByteBuffer.wrap(readBytes(cb));
```

Comment:
The protocol allows null values in a [bytes map]. We should use `readValue` instead of `readBytes` to support that.